components/eventing-controller/pkg/backend/jetstreamv2/jetstream_internal_unit_test.go

```diff
@@ -406,6 +208,99 @@ func Test_SyncNATSConsumers(t *testing.T) {
 	}
 }
 
+// Test_SyncNATSConsumers_ForErrors test the syncNATSConsumers for right error handling.
+func Test_SyncNATSConsumers_ForErrors(t *testing.T) {
+	// given
+	callback := func(m *nats.Msg) {}
+	subWithOneType := subtesting.NewSubscriptionWithOneType()
+	jsCleaner := &cleaner.JetStreamCleaner{}
+	defaultLogger, err := logger.New(string(kymalogger.JSON), string(kymalogger.INFO))
+	require.NoError(t, err)
+
+	testCases := []struct {
+		name             string
+		jetStreamContext *jetStreamContextStub
+		wantError        error
+	}{
+		{
+			name: "ConsumerInfo's not found error should be ignored",
+			jetStreamContext: &jetStreamContextStub{
+				consumerInfoError: nats.ErrConsumerNotFound,
+				consumerInfo:      nil,
+
+				addConsumer: &nats.ConsumerInfo{Config: nats.ConsumerConfig{MaxAckPending: DefaultMaxInFlights}},
+
+				subscribe: &nats.Subscription{},
+			},
+			wantError: nil,
```

Comment:
Shouldn't we test here that if the consumer is not found it was created ? Therefore that addConsumer was called ?